crates/relayer/src/chain/ckb4ibc/message/chan.rs

```diff
@@ -0,0 +1,497 @@
+use ckb_ics_axon::message::MsgChannelOpenAck as CkbMsgChannelOpenAck;
+use ckb_ics_axon::message::MsgChannelOpenConfirm as CkbMsgChannelOpenConfirm;
+use ckb_ics_axon::message::MsgChannelOpenInit as CkbMsgChannelOpenInit;
+use ckb_ics_axon::message::MsgChannelOpenTry as CkbMsgChannelOpenTry;
+use ckb_ics_axon::message::MsgRecvPacket as CkbMsgRecvPacket;
+use ckb_ics_axon::message::MsgType;
+use ckb_ics_axon::object::Packet as CkbPacket;
+use ckb_ics_axon::object::{ChannelCounterparty, Ordering as CkbOrdering, State as CkbState};
+use ckb_jsonrpc_types::TransactionView;
+use ckb_types::core::{Capacity, TransactionView as CoreTransactionView};
+use ckb_types::packed::{CellOutput, Script, WitnessArgs};
+use ckb_types::prelude::{Builder, Entity, Pack};
+use ibc_relayer_types::core::ics04_channel::channel::{ChannelEnd, Order, State};
+use ibc_relayer_types::core::ics04_channel::msgs::recv_packet::MsgRecvPacket;
+use ibc_relayer_types::core::ics04_channel::msgs::{
+    chan_close_init::MsgChannelCloseInit, chan_open_ack::MsgChannelOpenAck,
+    chan_open_confirm::MsgChannelOpenConfirm, chan_open_init::MsgChannelOpenInit,
+    chan_open_try::MsgChannelOpenTry,
+};
+use ibc_relayer_types::core::ics04_channel::packet::Packet;
+use ibc_relayer_types::core::ics24_host::identifier::ChannelId;
+use ibc_relayer_types::core::ics24_host::identifier::PortId;
+use itertools::Itertools;
+
+pub fn convert_chan_open_init_to_tx<C: MsgToTxConverter>(
+    msg: MsgChannelOpenInit,
+    converter: &C,
+) -> Result<TransactionView, Error> {
+    let old_connection_cell = converter.get_ibc_connections();
+    let next_channel_num = old_connection_cell.next_channel_number;
+    let mut new_connection_cell = old_connection_cell.clone();
+    new_connection_cell.next_channel_number += 1;
+
+    let ibc_channel_end = convert_channel_end(msg.channel, msg.port_id.clone(), next_channel_num)?;
+    let ibc_channel_end_encoded = get_encoded_object(ibc_channel_end);
+
+    let old_connection_encoded = get_encoded_object(old_connection_cell);
+    let new_connection_encoded = get_encoded_object(new_connection_cell);
+
+    let envelope = Envelope {
+        msg_type: MsgType::MsgChannelOpenInit,
+        content: rlp::encode(&CkbMsgChannelOpenInit {}).to_vec(),
+    };
+    let envelope_witness = get_encoded_object(envelope).witness;
+    let data_capacity = Capacity::zero()
+        .safe_add(32u32)
+        .map_err(|_| Error::capacity())?;
+    let packed_tx = CoreTransactionView::new_advanced_builder()
+        .input(converter.get_ibc_connections_input())
+        .output(
+            CellOutput::new_builder()
+                .lock(
+                    Script::new_builder()
+                        .code_hash(converter.get_config().connection_code_hash.pack())
+                        .args(converter.get_config().connection_args.pack())
+                        .build(),
+                )
+                .build_exact_capacity(data_capacity.clone())
```

Comment:
it's better to wrap it and other similar code as a utility function to use, for example, add a new `utils.rs` file to do so